documentation/tutorial.md

```diff
@@ -109,6 +109,49 @@ some functions:
 3. parse_item(): function which will be called to handle response
    downloaded by Crawly. It must return the `Crawly.ParsedItem` structure.
 
+As it is, the init function returns an empty array for `start_urls` which we will
+fix now. If you already know what urls, you want to start crawling, you can place them in
+the array, else you can pick the urls you want to start with from a particular page
+on your website of interest, in this case `https://books.toscrape.com`.
+
+Go ahead and modify your `init` to look this way.
+
+```elixir
+  @impl Crawly.Spider
+  def init() do
+    # Get all the cateory URLs to use as our starting point
+    response = Crawly.fetch("https://books.toscrape.com/catalogue/category/books_1/index.html")
+
+    {:ok, document} = Floki.parse_document(response.body)
+
+    # Extract product category URLs
+    product_categories_urls =
+      document
+      |> Floki.find("div.side_categories")
+      |> Floki.find("ul.nav-list > li > ul > li")
+      |> Floki.find("a")
+      |> Floki.attribute("href")
+      |> Enum.uniq()
+      |> Enum.map(&build_absolute_url/1)
+
+    [
+      start_urls: product_categories_urls
+    ]
+  end
+```
+
+We also need to add the `build_absolute_url/1` function which is used to build an absolute url from relative urls
+
+```elixir
+  defp build_absolute_url(url) do
```

Comment:
This function is already available in `Crawly.Utils`

Reply:
Done